### ISSUE TITLE
"Implement Cassandra index in terms of index interface" broke querying with cassandra index

### ISSUE THREAD

**Issue by arodland** [pyr/cyanite#209]
https://github.com/pyr/cyanite/commit/fc85da51dbb15060139d945cd377e43d69c3cc31 changed `prefixes` to return something without a "path" key, which is inconsistent with other indexes, and causes a NPE like so: https://gist.github.com/arodland/2218ee4af35b2a38adb40263f86d54d4 .

Adding `:path path` to `prefix-info` fixes the NPE, but is probably far from the best fix.

Relatedly, there is https://github.com/brutasse/graphite-cyanite/pull/21 which relates to graphite-api being broken by the same dropping of `path` at the API layer.

**Comment by arodland** [comment]
(forgive any lack of precision, I don't speak Clojure so I'm stumbling around in the dark here a bit.)

**Comment by ifesdjeen** [comment]
Fixed in last master.

**Comment by ifesdjeen** [comment]
Thanks for reporting! The first implementation was a bit overtuned. We've already done some work on improving it and working on more improvements.

**Comment by arodland** [comment]
Sorry if I wasn't clear. It's broken for me on current master . I get Internal Server Error / NPE for any render other than ones that return no data points.

**Comment by ifesdjeen** [comment]
I'll take a closer look at it tonight. I've been testing against grafana proxy, which might be a bit different setting. I'll take a closer look at how graphite-cyanite integrates.

**Comment by pauzed** [comment]
Yes, I was getting similar problems using graphite-api, and this made everything happier:

diff --git a/src/io/cyanite/index/cassandra.clj b/src/io/cyanite/index/cassandra.clj
index e918fe9..74a1a27 100644
--- a/src/io/cyanite/index/cassandra.clj
+++ b/src/io/cyanite/index/cassandra.clj
@@ -52,7 +52,8 @@

 (defn prefix-info
   [pattern-length [path length leaf]]
-  {:text          (last (split path #"\."))
+  {:path         path
+   :text         (last (split path #"\."))
    :id            path
    :allowChildren (not leaf)
    :expandable    (not leaf)
